### DISCUSSION TITLE
IP Address of client behind caprover

### DISCUSSION THREAD

**Discussion by emirefek** [caprover/caprover#1527] [General]
Hello, how we can access real client address inside caprover apps? I tried "X-FORWARDED-FOR" header for exposed ports but they didn't worked. What should I do? 

I found a already created issue https://github.com/caprover/caprover/issues/853 : If needed this discussion can be closed. But I think being able to get real client ip address without hussle will be awesome.

**Comment by githubsaturn** [comment]
exposed ports bypass nginx, hence you'll probably end up seeing the docker internal IP....

The issue that you linked is correct. This is a limitation of Docker. 

But if you don't used mapped port and use the regular Container HTTP port, so that the request passes through nginx, you'll be able to use `X-FORWARDED-FOR`
